### PR TITLE
Zephyr fixes

### PR DESCRIPTION
**Pull request by de-nordic**
The PR is series of fixes that to the Zephyr fork that:
 - change slot selection logic of `img_mgmt_find_best_area_id`, renames it to `mgm_mgmt_get_unused_area_id` and adds single parameter; now the function will accept as -1 as a parameter that request any upgreadable slot area_id, selected from first two slots, or positive number that is direct slot request; direct slot request may not be successful for first two slots as they are checked for usage, while no such checks are performed for all other slots.  All other slots are just checked for existence.
 - add missing check, of `img_mgmt_get_second_slot_area_id` return, in `img_mgmt_impl_erase_slot`;
 - changes the `img_mgmt_impl_write_image_data` to use the `area_id` from the state of mcumgr, so the currently selected for operation, instead of calling `img_mgmt_get_second_slot_area_id` (formerly `img_mgmt_find_best_area_id`) to figure it out again.

### CODE REVIEW

**Comment by de-nordic** [comment]
Please take a look @mlaz, @nvlsianpu , @utzig, Thanks!

**Review by utzig — APPROVED** [review]

**Comment by de-nordic** [comment]
@utzig Thanks for quick review!

**Comment by utzig** [comment]
> @utzig Thanks for quick review!

I will wait for @nvlsianpu approval!

**Review by sjanc — APPROVED** [review]

File: cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c
Comment (nvlsianpu) [inline cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c:127]: I'm afraid that this will made update impossible for DIRECT-XIP MCUboot compatible mode, when either primary or secondary slot might be the update target.

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

**Comment by de-nordic** [comment]
@utzig @sjanc Sorry for inconvenience, can you look again?

**Review by nvlsianpu — APPROVED** [review]

**Review by utzig — APPROVED** [review]
LGTM

**Pull request merged**
